File: src/main/java/instance/common/OSInit.java
package instance.common;

import cloud.api.InstanceConfiguration;
import cloud.common.NodeConfiguration;
import instance.Node;
import se.sics.kompics.Init;
import se.sics.kompics.address.Address;

import java.util.List;

/**
 * 
 * @author Amir Moulavi
 * @date 2011-03-18
 *
 */

public class OSInit extends Init {

	private InstanceConfiguration instanceConfiguration;
	
	public OSInit(InstanceConfiguration instanceConfiguration) {
		this.instanceConfiguration = instanceConfiguration;
	}
	
	public List<Block> getBlocks() {
		return instanceConfiguration.getBlocks();
	}

	public Address getCloudProviderAddress() {
		return instanceConfiguration.getCloudProviderAddress();
	}
	
	public Address getSelfAddress() {
		return instanceConfiguration.getSelfAddress();
	}

	public Node getNodeInfo() {
		return instanceConfiguration.getNode();
	}
	
	public NodeConfiguration getNodeConfiguration() {
		return instanceConfiguration.getNodeConfiguration();
	}
}

File: src/main/java/instance/common/OSInit.scala
package instance.common

import se.sics.kompics.Init
import cloud.api.InstanceConfiguration
import se.sics.kompics.address.Address
import instance.Node
import cloud.common.NodeConfiguration
import java.util.{List => JList}
/**
 * @author Amir Moulavi
 * @date 2011-07-16
 *
 */
class OSInit(instanceConfiguration:InstanceConfiguration) extends Init {

  def getBlocks: JList[Block] = instanceConfiguration.getBlocks

  def getCloudProviderAddress: Address = instanceConfiguration.getCloudProviderAddress

  def getSelfAddress: Address = instanceConfiguration.getSelfAddress

  def getNodeInfo: Node = instanceConfiguration.getNode

  def getNodeConfiguration: NodeConfiguration = instanceConfiguration.getNodeConfiguration

}